File: easy/108. Convert Sorted Array to Binary Search Tree/main.java
// 108. Convert Sorted Array to Binary Search Tree
class Solution {
    public TreeNode sortedArrayToBST(int[] nums) {
        return build(nums, 0, nums.length - 1);
    }
    
    private TreeNode build(int[] nums, int left, int right) {
        if (left > right) return null;
        int mid = left + (right - left) / 2;
        TreeNode root = new TreeNode(nums[mid]);
        root.left = build(nums, left, mid - 1);
        root.right = build(nums, mid + 1, right);
        return root;
    }
}

File: easy/108. Convert Sorted Array to Binary Search Tree/main.rs
// 108. Convert Sorted Array to Binary Search Tree
use std::rc::Rc;
use std::cell::RefCell;

impl Solution {
    pub fn sorted_array_to_bst(nums: Vec<i32>) -> Option<Rc<RefCell<TreeNode>>> {
        Self::build(&nums, 0, nums.len() as i32 - 1)
    }
    
    fn build(nums: &Vec<i32>, left: i32, right: i32) -> Option<Rc<RefCell<TreeNode>>> {
        if left > right {
            return None;
        }
        let mid = left + (right - left) / 2;
        let mut node = TreeNode::new(nums[mid as usize]);
        node.left = Self::build(nums, left, mid - 1);
        node.right = Self::build(nums, mid + 1, right);
        Some(Rc::new(RefCell::new(node)))
    }
}